sc2reader/readers.py

```diff
@@ -170,7 +170,7 @@ def __call__(self, data, replay):
                 observe=p[7],
                 result=p[8],
                 working_set_slot=p[9] if replay.build >= 24764 else None,
-                hero=p[10] if replay.build >= 34784 else None,
+                hero=p[10] if replay.build >= 34784 and 10 in p else None,
```

Comment:
Can you add a note here saying that hero only seems included in Heroes and not Starcraft replays? Any exceptional conditions should have reason noted in case it needs to be revisited later.